prometheus_client/registry.py

```diff
@@ -75,6 +75,16 @@ def collect(self):
             for metric in collector.collect():
                 yield metric
 
+    def reset(self):
+        """Clears value for each metric in each collector in the registry."""
+        collectors = None
+        with self._lock:
+            collectors = self._collector_to_names
+            for collector in collectors:
+                if hasattr(collector, 'reset'):
+                    collector.reset()
+
```

Comment:
2 comments on this - 
1. Ideally we wouldn't lock for the whole loop, but since we are mutating the collects we can't just do a copy like collect does above.  Maybe we could copy and then replace with the copy but needs thought.
2. I don't like the `hasattr(collector, 'reset')` because it makes behavior inconsistent, but maybe I've been thinking about this too long?